test/QueryPlannerTestHelpers.h

```diff
@@ -93,19 +93,32 @@ inline auto TextIndexScanForWord = [](Variable textRecordVar,
   return RootOperation<::TextIndexScanForWord>(AllOf(
       AD_PROPERTY(::TextIndexScanForWord, getResultWidth,
                   Eq(1 + word.ends_with('*'))),
-      AD_PROPERTY(::TextIndexScanForWord, textRecordVar, Eq(textRecordVar)),
-      AD_PROPERTY(::TextIndexScanForWord, word, word)));
+      AD_PROPERTY(::TextIndexScanForWord, getTextRecordVar, Eq(textRecordVar)),
+      AD_PROPERTY(::TextIndexScanForWord, getWord, word)));
 };
 
 inline auto TextIndexScanForEntity =
     [](Variable textRecordVar, std::variant<Variable, std::string> entity,
        string word) -> QetMatcher {
-  return RootOperation<::TextIndexScanForEntity>(AllOf(
-      AD_PROPERTY(::TextIndexScanForEntity, getResultWidth,
-                  Eq(2 + std::holds_alternative<Variable>(entity))),
-      AD_PROPERTY(::TextIndexScanForEntity, textRecordVar, Eq(textRecordVar)),
-      AD_PROPERTY(::TextIndexScanForEntity, entity, Eq(entity)),
-      AD_PROPERTY(::TextIndexScanForEntity, word, word)));
+  if (std::holds_alternative<Variable>(entity)) {
+    return RootOperation<::TextIndexScanForEntity>(
+        AllOf(AD_PROPERTY(::TextIndexScanForEntity, getResultWidth,
+                          Eq(2 + std::holds_alternative<Variable>(entity))),
+              AD_PROPERTY(::TextIndexScanForEntity, getTextRecordVar,
+                          Eq(textRecordVar)),
+              AD_PROPERTY(::TextIndexScanForEntity, getEntityVariable,
+                          std::get<Variable>(entity)),
+              AD_PROPERTY(::TextIndexScanForEntity, getWord, word)));
```

Comment:
You can add the correspoinding tests for `isFixedEntity`  (true in one case and false in the other).
Add a //TODO<joka921> Implement `AD_THROWING_PROPERTY(....., Exception matcher)` and use it here.